compose/ui/ui-test-junit4/src/skikoMain/kotlin/androidx/compose/ui/test/junit4/ComposeRootRegistry.skiko.kt

```diff
@@ -29,18 +29,22 @@ internal class ComposeRootRegistry {
     private val lock = Any()
     private val roots = mutableSetOf<SkiaRootForTest>()
 
+    // keep the function references in a dedicated val to make sure the instance always the same
+    private val _onRootCreated = ::onRootCreated
+    private val _onRootDisposed = ::onRootDisposed
+
     /**
      * Returns if the registry is setup to receive registrations from [SkiaRootForTest]s
      */
     val isSetUp: Boolean
-        get() = SkiaRootForTest.onRootCreatedCallback == ::onRootCreated
+        get() = SkiaRootForTest.onRootCreatedCallback == _onRootCreated
```

Comment:
cc @MatkovIvan JFYI: unfortunately, not all targets reuse a the same reference here. 
Apparently, k/wasm creates a new instance every time and the check is always false.

Reply:
Let's just store `isSetUp` as a boolean flag then

Reply:
ACK